src/l2_packet/l2_packet_zephyr.c

```diff
@@ -41,35 +44,63 @@ int l2_packet_get_own_addr(struct l2_packet_data *l2, u8 *addr)
 int l2_packet_send(struct l2_packet_data *l2, const u8 *dst_addr, u16 proto,
 		   const u8 *buf, size_t len)
 {
-	int ret;
+	int ret = -1;
+	const struct device *device = NULL;
+	const struct zep_wpa_supp_dev_ops *dev_ops = NULL;
 
 	if (l2 == NULL) {
 		return -1;
 	}
 
+	/* Get the device from the interface */
+	device = net_if_get_device(l2->iface);
+	if (!device) {
+		wpa_printf(MSG_ERROR, "l2_packet_send: Failed to get device");
+		return -1;
+	}
+
+	/* Get driver operations */
+	dev_ops = get_dev_ops(device);
+	if (!dev_ops) {
+		wpa_printf(MSG_ERROR, "l2_packet_send: Failed to get driver ops");
+		return -1;
+	}
+
+#ifdef CONFIG_WIFI_NM_WPA_SUPPLICANT_L2_PKT_DIRECT
```

Comment:
Maybe protect the entire set of changes (i.e. starting from line 55) with  `CONFIG_WIFI_NM_WPA_SUPPLICANT_L2_PKT_DIRECT` ? 

(Same comment for changes in other functions too).